### DISCUSSION TITLE
Installed PV power in each building

### DISCUSSION THREAD

**Discussion by irenemansoborras** [architecture-building-systems/CityEnergyAnalyst#3202] [Q&A] [answered]
Hello,
I have some questions in relation with how to figure out the installed PV capacity that CEA considers for each building. 

The area covered by PV panels is known. Nevertheless, in the PV panel specification, the area for panels is not defined (just the lenght) and the capacity is given in minimum and maximum (W). 

![image](https://user-images.githubusercontent.com/105488775/178946836-e4a867a3-125e-4614-8814-2583d2be34d3.png)

I would like to know how to figure out with the PV installed capacity in each building, if you could help me.
Thank you so much, 

Irene

**Comment by MatNif** [comment, marked as answer]
Hi Irene



The script that is called when launching the PV panel script is "photovoltaic.py" (normally located in your _User_/Documents/CityEnergyAnalyst/cea/technologies/solar folder). The function description in that file reads:

`' This function first determines the surface area with sufficient solar radiation, and then calculates the optimal tilt angles of panels at each surface location. The panels are categorized into groups by their surface azimuths, tilt angles, and global irradiation. In the last [section of the function], electricity generation from PV panels of each group is calculated [and summed up]. '`

I hope that answers your question regarding the methodology.



As for your second question, the nominal capacity is unfortunately not output directly. 'photovotalic.py' line 691 calculates the nominal capacity as:

`P_nominal_W = total_module_area_m2 * (constants.STC_RADIATION_Wperm2 * nominal_efficiency)`

`total_module_area_m2`: This can be found in your output files (i.e. under "../_test_case_/_test_scenario_/outputs/data/potentials/solar/PV_total_buildings.csv")
`constants.STC_RADIATION_Wperm2`: STC stands for "standard testing conditions" and a default value of 1000 W/m^2 is chosen for this
`nominal_efficiency`: can be found in the table you made a screenshot of in your question above

The easiest way to get the capacity of the installed PV panels, would probably be to replicate the function above in a spreadsheet editor (e.g. Excel). 
I know this is not exactly the most convenient solution. But as mentioned in our last newsletter, we are a bit short handed in the developers team currently, so none of us will get around to making any changes to this file any time soon. 
If you would like to make some changes to the CEA scripts yourself, you're very welcome to [contribute to the code](https://www.cityenergyanalyst.com/video-tutorials/v/how-to-modify-cea-code-and-submit-a-pull-request).

**Reply by irenemansoborras** [reply]
Thank you for your answer Mathias

best,
Irene
